internal/kubernetes/operator/federatedauthentication/federatedauthentication.go

```diff
@@ -0,0 +1,175 @@
+package federatedauthentication
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/kubernetes/operator/resources"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store"
+	akoapi "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	akov2common "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/common"
+	akov2status "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/status"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20240530003/admin"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	ErrNoMatchingSAMLProvider = errors.New("failed to retrieve the SAML identity provider matching the legacy ID")
+)
+
+type AtlasFederatedAuthBuildRequest struct {
+	IncludeSecret                bool
+	IdentityProviderLister       store.IdentityProviderLister
+	ConnectedOrgConfigsDescriber store.ConnectedOrgConfigsDescriber
+	ProjectStore                 store.OperatorProjectStore
+	IdentityProviderDescriber    store.IdentityProviderDescriber
+	ProjectName                  string
+	OrgID                        string
+	ProjectID                    string
+	FederatedSettings            *atlasv2.OrgFederationSettings
+	Version                      string
+	TargetNamespace              string
+	Dictionary                   map[string]string
+}
+
+const credSecretFormat = "%s-credentials"
+
+// BuildAtlasFederatedAuth builds an AtlasFederatedAuth resource.
+func BuildAtlasFederatedAuth(br *AtlasFederatedAuthBuildRequest) (*akov2.AtlasFederatedAuth, error) {
+	orgConfig, err := getOrgConfig(br)
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := getAtlasFederatedAuthSpec(*br, orgConfig)
+	if err != nil {
+		return nil, err
+	}
+	federatedAuth := &akov2.AtlasFederatedAuth{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "atlas.mongodb.com/v1",
+			Kind:       "AtlasFederatedAuth",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      resources.NormalizeAtlasName(fmt.Sprintf("%s-%s", br.ProjectName, br.FederatedSettings.GetId()), br.Dictionary),
+			Namespace: br.TargetNamespace,
+		},
+		Spec: *spec,
+		Status: akov2status.AtlasFederatedAuthStatus{
+			Common: akoapi.Common{
+				Conditions: []akoapi.Condition{},
+			},
+		},
+	}
+	return federatedAuth, nil
+}
+
+// getOrgConfig retrieves the organization configuration for the AtlasFederatedAuth resource.
+func getOrgConfig(br *AtlasFederatedAuthBuildRequest) (*atlasv2.ConnectedOrgConfig, error) {
+	return br.ConnectedOrgConfigsDescriber.GetConnectedOrgConfig(&atlasv2.GetConnectedOrgConfigApiParams{
+		FederationSettingsId: br.FederatedSettings.GetId(),
+		OrgId:                br.OrgID,
+	})
+}
+
+// getAtlasFederatedAuthSpec returns the spec for AtlasFederatedAuth.
+func getAtlasFederatedAuthSpec(br AtlasFederatedAuthBuildRequest, orgConfig *atlasv2.ConnectedOrgConfig) (*akov2.AtlasFederatedAuthSpec, error) {
```

Comment:
```suggestion
func atlasFederatedAuthSpec(br AtlasFederatedAuthBuildRequest, orgConfig *atlasv2.ConnectedOrgConfig) (*akov2.AtlasFederatedAuthSpec, error) {
```

It is not idiomatic to use the prefix `Get` in golang https://go.dev/doc/effective_go#Getters